cluster-autoscaler/core/static_autoscaler.go

```diff
@@ -85,30 +86,30 @@ func (a *StaticAutoscaler) RunOnce(currentTime time.Time) {
 	readyNodes, err := readyNodeLister.List()
 	if err != nil {
 		glog.Errorf("Failed to list ready nodes: %v", err)
-		return
+		return errors.ToAutoscalerError(errors.ApiCallError, err)
 	}
 	if len(readyNodes) == 0 {
 		glog.Error("No ready nodes in the cluster")
 		scaleDown.CleanUpUnneededNodes()
-		return
+		return nil
```

Comment:
Yes, I don't think having an empty cluster is an autoscaler error, so I'm returning nil here. I will update log levels accordingly. 

So far I've only did enough in core/ to get this PR to compile, so I could quickly get feedback on the idea of introducing typed errors. We have a lot of places where returning error signal failure of some function, but not an actual error condition that should be reported as such (e.g. not scaling up due to reaching max cluster size is expected autoscaler behavior). So I will have to make a lot of changes all over the codebase and I wanted to get other opinions before I do so.